### PR TITLE
issue14 search result component and tests

### PR DESCRIPTION
**Pull request by hondahmd**
- search result components and related actions, reducers and thunks
- search result components tests

### CODE REVIEW

**Comment by adamgyulavari** [comment]
fix the conflicts, because this PR contains many things right now that are already in the master

**Comment by hondahmd** [comment]
@adamgyulavari Conflicts resolved

File: frontend-opal/src/reducers/searchResult.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const pairs = {
+  SEARCH: (state, action) => action.payload.results
+};
+
+function searchResult(state = {}, action) {
```
Comment (adamgyulavari) [inline frontend-opal/src/reducers/searchResult.js]: @hondahmd this is exactly the same in the other, let's just export the pairs and create the reducers in the root

File: frontend-opal/src/components/SearchResult/Result.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { View, FlatList, Text, Image } from 'react-native';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import styles from './styles/result';
+import tradeIcon from './icons/trade.png';
+import infoIcon from './icons/info.png';
```
[the remark this answers is not in the document]
Comment (hondahmd) [reply frontend-opal/src/components/SearchResult/Result.js]: @adamgyulavari actually they're in the same dir, but I've changed them to absolute paths anyway

**Comment by hondahmd** [comment]
@adamgyulavari all changes done

**Pull request merged**